### PR TITLE
curlie: 1.6.7 -> 1.6.9

### PR DESCRIPTION
**Pull request by terlar**
###### Description of changes

Updates curlie to version [1.6.9](https://github.com/rs/curlie/releases/tag/v1.6.9).

Also I noticed the version of curlie was not set correctly via the build arguments, so fixed that and added a version test.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ModdedGamers** [comment]
Result of `nixpkgs-review pr 167258` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>curlie</li>
  </ul>
</details>

**Review by pyrox0 — APPROVED** [review]

**Review by superherointj — APPROVED** [review]

File: pkgs/tools/networking/curlie/default.nix [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
-    sha256 = "sha256-uWLJWhsqJaLji2JSuVX8Vu929AdozhtAPwsqXdpEt84=";
+    hash = "sha256-3EKxuEpFm+lp2myMfymYYY9boSXGOF2iAdjtGKnjJK0=";
   };
 
   vendorSha256 = "sha256-tYZtnD7RUurhl8yccXlTIvOxybBJITM+it1ollYJ1OI=";
 
   doCheck = false;
 
+  ldflags = [
+    "-X main.version=${version}"
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/curlie/default.nix:19]:
```suggestion
    "-s" "-w" "-X main.version=${version}"
```

Comment (terlar) [reply pkgs/tools/networking/curlie/default.nix:19]: Are those defaults that was there before setting the `ldflags`?

Comment (SuperSandro2000) [reply pkgs/tools/networking/curlie/default.nix]: no

**Pull request merged**